### PR TITLE
Updating to oclif v2

### PR DESCRIPTION
**Pull request by em-c-rod**
Oclif has upgraded to v2 and will not be maintaining v1. Therefore, it behooves us to move to v2 for better sustainability and added functionality.

See Oclif's v2 reference blog post [here](https://oclif.io/blog/2022/01/12/announcing-oclif-v2).
See Oclif's migration documentation [here](https://github.com/oclif/core/blob/main/MIGRATION.md).

Signed-off-by: Emily Rodriguez <ecrodriguez@mm279976-pc.lan>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -94,6 +98,7 @@
       "registry": "https://www.npmjs.com/package/@mitre/saf",
       "message": "<%= config.name %> update available from <%= chalk.greenBright(config.version) %> to <%= chalk.greenBright(latest) %>."
     },
+    "topicSeparator": ":",
```
Comment (em-c-rod) [inline package.json:101]: `":"` is the default, but with v2 this can be changed and specified to be a space `" "`.

File: package.json [hunk 2]
```diff
@@ -42,6 +41,7 @@
     "moment": "^2.29.1",
     "mustache": "^4.2.0",
     "objects-to-csv": "^1.3.6",
+    "oclif": "^2",
```
Comment (em-c-rod) [inline package.json:44]: Is `oclif` necessary outside of `devDependencies`? Its functionality for us is replacing `oclif-dev` to do things such as call `yarn prepack` or `yarn version` in the scripts below.

File: package.json (hunk 1 above)
Comment (camdenmoors) [reply package.json:100]: @aaronlippold @ejaronne Do we want to switch this? It would mean breaking changes for every implementation of the CLI currently unless there's a way to support both

Comment (aaronlippold) [reply package.json:100]: space seems most natural to me

File: package.json [hunk 3]
```diff
@@ -80,10 +80,14 @@
   "license": "Apache-2.0",
   "main": "lib/index.js",
   "oclif": {
+    "main": "lib/index.js",
     "commands": "./lib/commands",
+    "additionalHelpFlags": ["-h"],
+    "additionalVersionFlags": ["-v"],
```
[the remark this answers is not in the document]
Comment (Amndeep7) [reply package.json:85]: i would keep them

File: package.json (hunk 1 above)
Comment (Amndeep7) [reply package.json:100]: it is not a breaking change according to the migration docs:

> NOTE: Using colons always works, even if you set the topicSeparator to spaces. This means that you can enable spaces in your CLI without introducing a breaking change to your users.

Can @em-c-rod double check to confirm that this is actually the case?

File: package.json (hunk 2 above)
Comment (Amndeep7) [reply package.json:44]: if it's not being used for anything else, then yeah let's move it to devdependencies

**Comment by Amndeep7** [comment]
please try to fix the deepsource findings - i'm not overly worried if they aren't, but it'd be nice.  it seems like it's complaining about unused variables that you're getting from decomposing arrays into variables.  i think you can probably use the spread operator to get around this issue.

File: package.json [hunk 4]
```diff
@@ -64,7 +64,7 @@
     "ts-mocha": "^9.0.2"
   },
   "engines": {
-    "node": ">=10.0.0"
+    "node": ">=12.0.0"
```
Comment (Amndeep7) [inline package.json:67]: why the change to the minimum node version?

**Review by Amndeep7 — CHANGES_REQUESTED** [review]
please try to fix the deepsource findings - i'm not overly worried if they aren't, but it'd be nice. it seems like it's complaining about unused variables that are a result of decomposing arrays into variables. i think it is possible to use the spread operator to get around this issue.

File: package.json (hunk 1 above)
Comment (em-c-rod) [reply package.json]: I changed it to space, tested locally, and added a test to the suite that uses the space instead of the ":" for a separator.

File: package.json (hunk 4 above)
Comment (em-c-rod) [reply package.json:67]: https://oclif.io/blog/2022/01/12/announcing-oclif-v2#node--12

**Comment by Amndeep7** [comment]
just to make sure, have you also tested if a manually built docker image still works?

**Comment by camdenmoors** [comment]
This seems to be breaking the help for top-level commands:

Screen Shot 2022-03-17 at 11 48 00 AM

**Comment by em-c-rod** [comment]
@camdenmoors Looks like it does have this help feature built in.
image

**Comment by em-c-rod** [comment]
@camdenmoors Fixed. You get the same output from all three of these commands: `saf convert`, `saf -h convert`, and `saf --help convert`.

**Review by camdenmoors — APPROVED** [review]
